model_signing/model.py

```diff
@@ -101,7 +101,10 @@ def sign(self, inputfn: Path, signaturefn: Path,
             oidc_token = self.get_identity_token()
             if not oidc_token:
                 raise ValueError("No identity token supplied or detected!")
-            print(f"identity-provider: {oidc_token.issuer}",
+            # Calling the private attribute IdentityToken._federated issuer is a workaround
+            # For earlier versions of sigstore-python (<3.0.0) that do not support the
+            # federated_issuer property.
+            print(f"identity-provider: {oidc_token._federated_issuer}",
```

Comment:
Is federated issuer always set? If a non-Dex token is used, what is this value?

Reply:
@woodruffw please advise

Reply:
@mayaCostantini this is the original issue + solution that @woodruffw told us should work https://github.com/sigstore/sigstore-python/issues/970#issuecomment-2108727868. So the right solution is probably `.expected_certificate_subject`

Reply:
With `sigstore-python` 3.0.0, `federated_issuer` should fall back to the "top-level" issuer if the Dex-specific federation claim isn't present. You can see that here:

https://github.com/sigstore/sigstore-python/blob/29c98ea7f58c3047b9b5e062af912f4523ff1514/sigstore/oidc.py#L208-L227

(Before 3.0.0, `federated_issuer` was called `expected_certificate_subject`, which was a misleading name. But they're the same API.)